### PR TITLE
2 lint commits

### PR DESCRIPTION
**Pull request by kevmoo**

### CODE REVIEW

File: dart/tool/java/generate_java.dart [hunk 1]
```diff
@@ -133,12 +133,12 @@ class Api extends Member with ApiParseUtil {
     for (var m in methods) {
       for (var a in m.args) {
         if (a.hasDocs) continue;
-        var t = types.firstWhere((t) => t.name == a.type, orElse: () => null);
```
Comment (kevmoo) [inline dart/tool/java/generate_java.dart]: @devoncarew  The `==` for compatible type lint is no nice. Take a look at this. Clearly there is some behavior here that was not working

Comment (devoncarew) [reply dart/tool/java/generate_java.dart]: Wow, good catch with this lint. Will run locally to see if I can determine the intent of this code.

**Comment by devoncarew** [comment]
Thanks! Will take a spin through this, but I'll probably want the bulk of these changes separate from the new/const change.

**Comment by devoncarew** [comment]
Yeah, I may not want the new/const changes, and separate PRs makes things easier to review independently.

**Comment by devoncarew** [comment]
Thanks for the PR! Largely landed in https://github.com/dart-lang/vm_service_drivers/pull/205 and https://github.com/dart-lang/vm_service_drivers/pull/200.

**Pull request closed without merging**